src/prefect/agent/kubernetes/agent.py

```diff
@@ -140,6 +156,16 @@ def generate_deployment_yaml(
 
         return yaml.safe_dump(deployment)
 
+    def heartbeat(self) -> None:
+        """
+        Write agent heartbeat by opening and closing a heartbeat file. This allows
+        liveness probes to check the agent's main process activity based on the
+        heartbeat file's last modified time.
+        """
+        os.makedirs(AGENT_DIRECTORY, exist_ok=True)
+
+        open("{}/heartbeat".format(AGENT_DIRECTORY), "w").close()
```

Comment:
:O this is all it takes to update the "last modified" time?

Reply:
Yeah since `w` clears